### PR TITLE
vulcan-prowler: select group of controls according to security-level

### PR DESCRIPTION
**Pull request by manelmontilla**
This PR modifies the vulcan-prowler check to accept a security level for an account and set the CIS Level to check for depending on that level.
Concretely:

- If the security level is 0 or 1, it will execute the controls included in the CIS Level 1.
- If the security is not specified or it is 2, it will execute the controls included in the CIS Level 2.

Apart from that there are some small improvements on the resources table and in the descriptions of the vulnerabilities.

### CODE REVIEW

File: cmd/vulcan-prowler/main.go [hunk 1]
```diff
@@ -44,26 +45,47 @@ var (
 	CISCompliance = report.Vulnerability{
 		Summary: "Compliance With CIS AWS Foundations Benchmark (BETA)",
 		Description: `<p>
+			The check did not received the security classification level
```
Comment (gguillemas) [inline cmd/vulcan-prowler/main.go:48]: `s/received/receive/`

File: cmd/vulcan-prowler/main.go [hunk 2]
```diff
@@ -44,26 +45,47 @@ var (
...
+			</p>
+			<p>
+		    The CIS AWS Foundations Benchmark provides prescriptive
+			guidance for configuring security options for a subset of Amazon Web
+			Services with an emphasis on foundational, testable, and architecture
+			agnostic settings. The services included in the scope are: IAM, Confing, 
+			CloudTrail, CloudWatch, SNS, S3 and VPC (Default).
+		</p>
+		<p>
+			The provided recommendations are provided as a result controls included in
```
Comment (gguillemas) [inline cmd/vulcan-prowler/main.go:87]: I would avoid saying provided twice in the same sentence. I also think this sentence is missing an "of".
Maybe we can leave it like: `Recommendations are provided in order to comply with all the controls required by the CIS Level 1.`

File: cmd/vulcan-prowler/main.go [hunk 3]
```diff
@@ -74,13 +96,59 @@ var (
...
+			</p>
+			<p>
+		    The CIS AWS Foundations Benchmark provides prescriptive
+			guidance for configuring security options for a subset of Amazon Web
+			Services with an emphasis on foundational, testable, and architecture
+			agnostic settings. The services included in the scope are: IAM, Confing, 
+			CloudTrail, CloudWatch, SNS, S3 and VPC (Default).
+		</p>
+		<p>
+			The provided recommendations are provided as a result of controls included in
```
Comment (gguillemas) [inline cmd/vulcan-prowler/main.go:117]: Same here as above.

File: cmd/vulcan-prowler/main.go [hunk 4]
```diff
@@ -74,13 +96,59 @@ var (
...
+		References: []string{
+			"https://d0.awsstatic.com/whitepapers/compliance/AWS_CIS_Foundations_Benchmark.pdf",
+			"https://github.com/toniblyx/prowler",
+			"https://www.cisecurity.org/benchmark/amazon_web_services/",
+		},
+		Score: report.SeverityThresholdMedium,
+	}
+
+	CISComplianceInfo = report.Vulnerability{
+		Summary: "Information about CIS AWS Foundations Benchmark (BETA)",
```
Comment (gguillemas) [inline cmd/vulcan-prowler/main.go:133]: `s/about/About/`

File: cmd/vulcan-prowler/main.go [hunk 5]
```diff
@@ -74,13 +96,59 @@ var (
...
+		Description: `<p>
+			     Information gathered by executing the CIS benchmark about the account.
```
Comment (gguillemas) [inline cmd/vulcan-prowler/main.go:135]: `s/about/on`

File: cmd/vulcan-prowler/main.go [hunk 6]
```diff
@@ -74,13 +96,59 @@ var (
...
+			"https://www.cisecurity.org/benchmark/amazon_web_services/",
+		},
+		Score: report.SeverityThresholdNone,
 	}
 )
 
 type options struct {
 	Region          string   `json:"region"`
 	Groups          []string `json:"groups"`
 	SessionDuration int      `json:"sessionDuration"` // In secs.
```
Comment (gguillemas) [inline cmd/vulcan-prowler/main.go:150]: I think we are doing `snake_case` for all the options in Vulcan. Maybe this is a good chance to change this or maybe we can do it in a separate PR.

File: cmd/vulcan-prowler/main.go [hunk 7]
```diff
@@ -136,86 +204,168 @@ func main() {
...
 	}
 
 	c := check.NewCheckFromHandler(checkName, run)
 	c.RunAndServe()
 }
 
-func addVulnsToState(state state.State, r *prowlerReport, alias string) {
-	v := CISCompliance
+func groupsFromOpts(opts options) ([]string, error) {
+	// If ROLFP level is specified then it defines the group to use.
```
Comment (gguillemas) [inline cmd/vulcan-prowler/main.go:251]: I would avoid referencing ROLFP anywhere, as this is the classification criteria that we have chosen in Adevinta. Externally I believe only the security level should be exposed, as it is independent from how each user of Vulcan actually calculates it.

File: cmd/vulcan-prowler/main.go [hunk 8]
```diff
@@ -136,86 +204,168 @@ func main() {
...
+	if opts.SecurityLevel == nil {
+		return opts.Groups, nil
+	}
+	level := *opts.SecurityLevel
+	if level < 0 || level > 2 {
+		return nil, errors.New("invalid rolfp level value")
```
Comment (gguillemas) [inline cmd/vulcan-prowler/main.go:257]: Same here with the ROLFP reference.

File: cmd/vulcan-prowler/main.go (hunk 1 above)
Comment (gguillemas) [reply cmd/vulcan-prowler/main.go:48]: Maybe we could also drop the "level" word like in the other descriptions.

Comment (manelmontilla) [reply cmd/vulcan-prowler/main.go:48]: Yes thanks

File: cmd/vulcan-prowler/main.go (hunk 2 above)
Comment (manelmontilla) [reply cmd/vulcan-prowler/main.go:87]: Makes absolutely sense.

File: cmd/vulcan-prowler/main.go (hunk 3 above)
Comment (manelmontilla) [reply cmd/vulcan-prowler/main.go:117]: True thanks

File: cmd/vulcan-prowler/main.go (hunk 4 above)
Comment (manelmontilla) [reply cmd/vulcan-prowler/main.go:133]: Yes thanks.

File: cmd/vulcan-prowler/main.go (hunk 5 above)
Comment (manelmontilla) [reply cmd/vulcan-prowler/main.go:135]: 👍

File: cmd/vulcan-prowler/main.go (hunk 6 above)
Comment (manelmontilla) [reply cmd/vulcan-prowler/main.go:150]: Yes makes sense, I'm doing it in this PR because, as you said, we are already using it to improve a little bit the check and not only for the taking into account the security level. I'm also modifying the local.toml.example file.

File: cmd/vulcan-prowler/main.go (hunk 7 above)
Comment (manelmontilla) [reply cmd/vulcan-prowler/main.go:251]: Yes true.

File: cmd/vulcan-prowler/main.go (hunk 8 above)
Comment (manelmontilla) [reply cmd/vulcan-prowler/main.go:257]: Yes thanks
